skorch/tests/test_net.py

```diff
@@ -1263,6 +1264,60 @@ def test_set_params_with_unknown_key_raises(self, net):
         # the intended message should be from sklearn side
         assert exc.value.args[0].startswith('Invalid parameter foo for')
 
+    def test_set_params_on_submodule(self, net_cls, module_cls):
+        class SuperModule(nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.sub_module = module_cls()
+
+            def forward(self, X):
+                return self.sub_module(X)
+
+        net = net_cls(SuperModule, module__sub_module__num_units=123)
+        net.initialize()
+        assert net.module_.sub_module.num_units == 123
```

Comment:
The following tests would fail:

```python
assert net.module_.sub_module.dense0.out_features == 123
assert net.module_.sub_module.dense1.in_features == 123
```

Setting `num_units` after `__init__` does not update the underlying modules. Is this the intended behavior?

Reply:
Exactly, this was my concern as well. This does not really solve the use-case from #107 (specifically, pay attention to the last sentence).